xds/internal/test/xds_server_integration_test.go

```diff
@@ -335,71 +299,169 @@ func listenerResourceWithSecurityConfig(t *testing.T, lis net.Listener) *v3liste
-				TransportSocket: transportSocket,
+				TransportSocket: ts,
+			},
+		},
+	}, nil
+}
+
+// inboundListenerResourceWithoutSecurityConfig returns a listener resource with
+// no security configuration, and name and address fields matching the passed in
+// net.Listener.
+func inboundListenerResourceWithoutSecurityConfig(lis net.Listener) (*v3listenerpb.Listener, error) {
+	return inboundListenerResourceWithTransportSocket(lis, nil)
+}
+
+// inboundListenerResourceWithMutualTLSConfig returns a listener resource with
+// security configuration pointing to the use of the file_watcher certificate
+// provider plugin, and name and address fields matching the passed in
+// net.Listener. The security configuration returned here corresponds to the use
+// of mTLS.
+func inboundListenerResourceWithMutualTLSConfig(lis net.Listener) (*v3listenerpb.Listener, error) {
+	return inboundListenerResourceWithTransportSocket(lis, &v3corepb.TransportSocket{
+		Name: "envoy.transport_sockets.tls",
+		ConfigType: &v3corepb.TransportSocket_TypedConfig{
+			TypedConfig: testutils.MarshalAny(&v3tlspb.DownstreamTlsContext{
+				RequireClientCertificate: &wrapperspb.BoolValue{Value: true},
+				CommonTlsContext: &v3tlspb.CommonTlsContext{
+					TlsCertificateCertificateProviderInstance: &v3tlspb.CommonTlsContext_CertificateProviderInstance{
+						InstanceName: "server-side-certificate-provider",
+					},
+					ValidationContextType: &v3tlspb.CommonTlsContext_ValidationContextCertificateProviderInstance{
+						ValidationContextCertificateProviderInstance: &v3tlspb.CommonTlsContext_CertificateProviderInstance{
+							InstanceName: "server-side-certificate-provider",
+						},
+					},
+				},
+			}),
+		},
+	})
+}
+
+// inboundListenerResourceWithTLSConfig returns a listener resource with
+// security configuration pointing to the use of the file_watcher certificate
+// provider plugin, and name and address fields matching the passed in
+// net.Listener. The security configuration returned here corresponds to the use
+// of TLS where the client does not present a certificate.
+func inboundListenerResourceWithTLSConfig(lis net.Listener) (*v3listenerpb.Listener, error) {
+	return inboundListenerResourceWithTransportSocket(lis, &v3corepb.TransportSocket{
+		Name: "envoy.transport_sockets.tls",
+		ConfigType: &v3corepb.TransportSocket_TypedConfig{
+			TypedConfig: testutils.MarshalAny(&v3tlspb.DownstreamTlsContext{
+				CommonTlsContext: &v3tlspb.CommonTlsContext{
+					TlsCertificateCertificateProviderInstance: &v3tlspb.CommonTlsContext_CertificateProviderInstance{
+						InstanceName: "server-side-certificate-provider",
+					},
+				},
+			}),
+		},
+	})
+}
+
+// xdsResourcesForClientSecurity creates xDS resources required on the client
+// side when security is enabled. This differs from the resources returned by
+// `e2e.DefaultClientResources` only in the cluster resource which contains the
+// security configuration. The mTLS argument controls whether the security
```

Comment:
If so, can this use `e2e.DefaultClientResources` and then modify them - or parameterize as needed?  I would like to minimize the amount of configs we have floating around for when things happen like "the router http2 filter is now required, so all tests fail unless you add it everywhere".

Reply:
Done.